src/Microsoft.AspNet.SignalR.Client/Connection.cs

```diff
@@ -975,6 +976,40 @@ private static string CreateQueryString(IDictionary<string, string> queryString)
             return String.Join("&", queryString.Select(kvp => kvp.Key + "=" + kvp.Value).ToArray());
         }
 
+        /// <summary>
+        /// returns the url without the query and the query string as two seperate items
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns>Item1:The url without the query string
+        /// Item2:The query string if there was one or null</returns>
+        private static string GetQueryStringFromUrl(string url)
+        {
+            if (url == null) throw new ArgumentNullException("url");
+            string queryString = null;
+            var firstIndex = url.IndexOf("?");
+            if (firstIndex > -1)
+            {
+                queryString = url.Substring(firstIndex + 1);
```

Comment:
just return instead of assigning - this is the pattern we use.

Reply:
or use ternary `return firstIndex > -1 ? queryString = url.Substring(firstIndex + 1) : null`